config-service-change-event-api/src/main/proto/org/hypertrace/config/change/event/v1/config_change_event_value.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package org.hypertrace.config.change.event.v1;
+
+import "google/protobuf/struct.proto";
+
+message ConfigChangeEventValue {
+  oneof event {
+    ConfigCreateEvent create_event = 1;
+    ConfigUpdateEvent update_event = 2;
+    ConfigDeleteEvent delete_event = 3;
+  }
+}
+
+message ConfigCreateEvent {
+  google.protobuf.Value created_config = 1;
```

Comment:
So our options here that I see are basically:
1. Use the actual types, and have a specific event per config type (doesn't seem feasible)
2. Use struct (which we know can represent all of our configs, since we use it that way internally today)
3. Use a byte and/or json string.

So given the last two options, what benefits do you see to the struct vs byte/json? drawbacks?

Reply:
Converting to byte/json is extra step for serialisation as well as deserialisation. That is the reason I went with option 2.